HelpMyStreetFE/HelpMyStreetFE/Services/AddressService.cs

```diff
@@ -100,21 +100,27 @@ public async Task<IEnumerable<LocationWithDistance>> GetLocationDetailsForUser(U
         {
             return await _memDistCache_LocationDistanceList.GetCachedDataAsync(async (cancellationToken) =>
             {
-                var locationsWithDistance = await _addressRepository.GetLocationsByDistance(_requestSettings.Value.ShiftRadius, user.PostalCode);
-
-                if (locationsWithDistance.Count() == 0)
+                if (user.PostalCode != null)
```

Comment:
This is sensible, but move the if statement ouside the call to `GetCachedDataAsync`.  Otherwise, the empty list will be cached and returned in future, even if the user subsequently gains a postcode.